packages/core/strapi/lib/types/core/attributes/component.d.ts

```diff
@@ -1,38 +1,34 @@
-import { Attribute, ConfigurableOption, MinMaxOption, PrivateOption, RequiredOption } from './base';
-import { GetAttributesValues } from './utils';
+import type { Common, Attribute } from '@strapi/strapi';
 
-export interface ComponentAttributeProperties<
+export interface ComponentProperties<
   // Targeted component
-  T extends Strapi.ComponentUIDs,
+  T extends Common.UID.Component,
   // Repeatable
   R extends boolean = false
 > {
   component: T;
   repeatable?: R;
 }
 
-export type ComponentAttribute<
+export type Component<
   // Targeted component
-  T extends Strapi.ComponentUIDs,
+  T extends Common.UID.Component = Common.UID.Component,
```

Comment:
Do you not think it's bad practice just using a one letter Generic? It's not clear to someone using `Component` what T is really imo? `TargetComponent` would be _loads clearer_...

Reply:
it would remove the need for self explanatory comments

Reply:
Yeah, I'm fine with a simple T when it's a one-liner with only one generic. More than one generic and I would suggest a T-prefix with a name.

Reply:
Just thinking out loud, but as the TS community is mainly using the Java legacy on the matter, I'm afraid it could be something that puts contributors off :confused:

Reply:
I don't think it will, popular libraries such as `react-query` & `redux` .etc clearly define their generics as actual variable names, just like actual variables a single letter adds to the cognitive load of reading code. If i'm honest, this is also something we as a company need to agree on in Typescript do's and don'ts moving forward as well to have a unified approach...

Reply:
Curious about your opinion on this one

```ts
export type GetBigIntegerValue<T extends Attribute.Attribute> = T extends BigInteger
  ? BigIntegerValue
  : never;
```
vs
```ts
export type GetBigIntegerValue<TAttribute extends Attribute.Attribute> =
  TAttribute extends BigInteger ? BigIntegerValue : never;
```

I personally find the first one a lot easier to read/understand but I might be biased. Looking for more opinion :eyes:

Reply:
Honestly the bottom 😅 if we zoom out, even though is a simpler case, some type functions could get quite complicated with their use of generics and i think it's about a blanket rule... as opposed to having a "you can use a single letter in some circumstances" which imo will cause ambiguity and therefore a misuse by accident.

Reply:
After looking online, asking and messing around, this is what I ended up doing:

- Type parameter (whether it's for interface, types, functions, ...) must start with `T` and be `PascalCased`
- For one-liner types with self-explanatory context, `T` is accepted
- For complex types or one-liners that require more context, a precise parameter name must be provided (`TSomething`)

Here is a list of examples to showcase the different scenario:
```ts
export type Dict<T> = Record<string, T>;

export type Prefix<TValue extends string, TPrefix extends Literal> = `${TPrefix}${TValue}`;

export type ExtractScope<TNamespace> = TNamespace extends `${string}${ColonsSeparator}${infer TScope}` ? TScope : never;

export type GetRelationValue<TAttribute extends Attribute.Attribute> = TAttribute extends Relation<
  infer _TOrigin,
  infer TRelationType,
  infer TTarget
>
  ? RelationValue<TRelationType, TTarget>
  : never;

export type GetTimeValue<T extends Attribute.Attribute> = T extends Time ? TimeValue : never;

export interface DynamicZoneProperties<TComponentsUIDs extends Common.UID.Component[]> {
  components: TComponentsUIDs;
}
```

I've (_well, credits to Alex in reality_) also found an eslint rule that should handle the naming rules for us: https://github.com/typescript-eslint/typescript-eslint/blob/main/packages/eslint-plugin/docs/rules/naming-convention.md#enforce-that-type-parameters-generics-are-prefixed-with-t
(It won't make the difference between `T` & `TSomething`, but at least it'll prevent other variable names such as `S`, `P`, `Tlowercase`, `lowercase` or `PascalCase`)

You can find all the related changes in this commit: [595a4c8](https://github.com/strapi/strapi/pull/16917/commits/595a4c885a1bd79e2bb4283a87984f33d7f857b0)

_**pinging @innerdvations @alexandrebodin @Marc-Roig for context, opinions & comments**_

Reply:
Great compromise – very clear – very nice 💯

Reply:
FYI: I didn't added the eslint rule for now as it would nuke the rest of our TS package, but we'll add this to our backlog

Reply:
(Watching JS and Christian working on this in Discord right now and these types are way more clear <3 )

Reply:
Sounds good to me!